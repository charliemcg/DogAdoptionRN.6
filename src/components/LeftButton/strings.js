export default {
  navigation: {
    userProfile: "UserProfile",
    signIn: "SignIn"
  }
};
