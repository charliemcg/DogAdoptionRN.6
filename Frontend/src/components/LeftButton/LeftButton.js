import React, {Component} from 'react';
import {connect} from 'react-redux';
import IconAwesome from 'react-native-vector-icons/FontAwesome';
import IconAwesome5 from 'react-native-vector-icons/FontAwesome5';
import colors from '../../colors';
import strings from './strings';

class LeftButton extends Component {
  render() {
    //return a log in icon when user is signed out. Return a user icon when user signed in.
    return this.props.signedIn ? (
      <IconAwesome5
        style={{paddingRight: 10, paddingLeft: 8}}
        name="user-circle"
        size={30}
        color={colors.contrast}
        onPress={() => {
          this.props.navigation.navigate(strings.navigation.userProfile);
        }}
      />
    ) : (
      <IconAwesome
        style={{paddingRight: 10, paddingLeft: 8}}
        name="sign-in"
        size={30}
        color={colors.contrast}
        onPress={() => {
          this.props.navigation.navigate(strings.navigation.signIn);
        }}
      />
    );
  }
}

const mapStateToProps = state => {
  return {
    signedIn: state.signedIn,
  };
};

export default connect(mapStateToProps, null)(LeftButton);
