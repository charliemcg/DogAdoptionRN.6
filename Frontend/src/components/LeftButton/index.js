import LeftButton from "./LeftButton";
export default LeftButton;
