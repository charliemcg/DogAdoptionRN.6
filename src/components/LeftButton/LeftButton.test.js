import "react-native";
import React from "react";
import LeftButton from "./LeftButton";
import renderer from "react-test-renderer";
import store from "../../store";

test("LeftButton snapShot", () => {
  const snap = renderer.create(<LeftButton store={store} />).toJSON();

  expect(snap).toMatchSnapshot();
});
