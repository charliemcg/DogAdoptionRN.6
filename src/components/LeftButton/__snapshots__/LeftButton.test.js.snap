// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftButton snapShot 1`] = `
<Text
  allowFontScaling={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#37474f",
        "fontSize": 30,
      },
      Object {
        "paddingLeft": 8,
        "paddingRight": 10,
      },
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  ÔÇê
</Text>
`;
